src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -4169,10 +4172,11 @@ private XamlLazyApplyBlockIIndentedStringBuilder CreateApplyBlock(IIndentedStrin
 
 			if (appliedType != null)
 			{
-				if (!_xamlAppliedTypes.TryGetValue(appliedType, out var appliedTypeIndex))
+				var globalizedFullyQualifiedAppliedType = GetGlobalizedTypeName(appliedType.ToString());
+				if (!_xamlAppliedTypes.TryGetValue(globalizedFullyQualifiedAppliedType, out var appliedTypeIndex))
```

Comment:
I'm not sure if this has the same semantics because two symbols might not be comparing equal but the corresponding fully qualified names are the same. @jeromelaban Is the change good for this specific case?

Reply:
Also let me know if you have a better fix. I'm very concerned about having a state in the generator in the first place.

Reply:
Qualified types by name is good enough. Having a state is mandatory at this time, until metadata updates handle more changes such as lambda parameter changes (let's hope .NET 8).

Reply:
@jeromelaban Is it the last item on https://github.com/dotnet/roslyn/issues/59264 ?

> Changing parameters on lambdas and local functions themselves

Reply:
It is exclusively the last item, yes.